File: FRDIntentDemo/FirstViewController.m
//
//  FirstViewController.m
//  FRDIntent
//
//  Created by GUO Lin on 9/13/16.
//  Copyright © 2016 Douban Inc. All rights reserved.
//

#import "FirstViewController.h"


@interface FirstViewController ()

@property (nonatomic, strong) NSDictionary<NSString *, id> *data;

@end

@implementation FirstViewController

- (id)initWithExtras:(NSDictionary<NSString *,id> *)extras {
  self = [super init];
  if (self) {
    _data = extras;
  }
  return self;
}

- (void)viewDidLoad {
  [super viewDidLoad];
  self.title = @"First View Controller";
  self.view.backgroundColor = [UIColor whiteColor];

  UILabel *numberLabel = [[UILabel alloc] init];
  numberLabel.frame = CGRectMake(20, 100, self.view.bounds.size.width - 40, 400);
  numberLabel.textAlignment = NSTextAlignmentCenter;
  numberLabel.numberOfLines = 0;
  if (self.data) {
    numberLabel.text = [NSString stringWithFormat:@"%@", self.data];
  }
  [self.view addSubview:numberLabel];
}

@end

File: FRDIntentDemo/FirstViewController.swift
//
//  FirstViewController.swift
//  FRDIntent
//
//  Created by GUO Lin on 8/25/16.
//  Copyright © 2016 Douban Inc. All rights reserved.
//

import UIKit
import FRDIntent

class FirstViewController: UIViewController, IntentReceivableController {

  var data: [String: Any]?

  convenience required init(extra: [String: Any]?) {
    self.init(data: extra)
  }

  init(data: [String: Any]?) {
    self.data = data
    super.init(nibName: nil, bundle: nil)
  }

  required init?(coder aDecoder: NSCoder) {
    fatalError("init(coder:) has not been implemented")
  }

  override func viewDidLoad() {
    super.viewDidLoad()
    title = "FirstViewController"
    view.backgroundColor = UIColor.whiteColor()
    let numberLabel = UILabel()
    numberLabel.frame = CGRect(x: 20, y: 100, width: view.bounds.size.width - 40, height: 500)
    numberLabel.textAlignment = .Center
    numberLabel.numberOfLines = 0
    if let data = data {
      numberLabel.text = "\(data)"
    }
    view.addSubview(numberLabel)
  }

}
